test/ComposerRequireCheckerTest/NodeVisitor/UsedSymbolCollectorFunctionalTest.php

```diff
@@ -128,6 +129,26 @@ public function testWontCollectAnyUsageTypes(): void
         );
     }
 
+    public function testWillCollectPropertyTypes(): void
+    {
+        $this->traverseStringAST('<?php class Foo { public My\ReturnType $foo; }');
+
+        self::assertSameCollectedSymbols(
+            ['My\ReturnType'],
+            $this->collector->getCollectedSymbols()
+        );
+    }
+
+    public function testWillCollectUseTypes(): void
+    {
+        $this->traverseStringAST('<?php use My\ReturnType;');
```

Comment:
IMO this is not to be implemented.

The problem is that `use` statements can be partial symbols.

For example, in this case, `Doctrine\ORM\Mapping` is not a defined symbol:

```php
use Doctrine\ORM\Mapping as ORM;

// ...

/** @ORM\Entity() */
class MyEntity {
    // ...
```

I use this very often also with functions:

```php
use Psl\Type;

// ...

Type\object(MyClass:class)
    ->assert($value);
```

Reply:
Fair enough, I'll remove the "use" part from this PR, but could you perhaps give me some hints on what would be the best way to implement symbols used in annotations?

Reply:
> but could you perhaps give me some hints on what would be the best way to implement symbols used in annotations?

Perhaps we can do that for PHP 8.1 attributes? Annotations are not that problematic anymore, and people will slowly move away from them...

Reply:
I think they are still used quite a lot to tell static analyzers what types to infer, for example:
```php
/**
 * @var array<\Ramsey\Uuid>
 */
private array $uuids;
```

In such a case if ramsey/uuid is not installed the checker should fail, no?

Reply:
Ah, that stuff: there are tools for that in https://github.com/phpDocumentor/ReflectionDocBlock (which allows you to avoid re-inventing the wheel).

Reply:
Ok, I'll see what I can do about it in a separate PR. Thanks!